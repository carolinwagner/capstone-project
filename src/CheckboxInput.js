import React, { forwardRef } from 'react'

const CheckboxInput = forwardRef(({ answerOptions, name }, ref) => {
  return (
    <div>
      {answerOptions.map((answerOption, innerIndex) => {
        const inputName = `${name}[${innerIndex}]`
        return (
          <div key={answerOption.name}>
            <input
              name={inputName}
              id={answerOption.name}
              value={answerOption.name}
              type="checkbox"
              ref={ref}
            />
            <label htmlFor={answerOption.name}>{answerOption.label}</label>
          </div>
        )
      })}
    </div>
  )
})

export default CheckboxInput
