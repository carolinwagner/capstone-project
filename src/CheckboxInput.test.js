import React from 'react'
import '@testing-library/jest-dom/extend-expect'
import renderer from 'react-test-renderer'
import CheckboxInput from './CheckboxInput'

const defaultQuestion = {
  answerOptions: [
    { label: 'a', name: 'a' },
    { label: 'b', name: 'b' },
    { label: 'c', name: 'd' },
    { label: 'd', name: 'd' },
  ],
}

describe('CheckboxInput', () => {
  it('renders CheckboxInput correctly', () => {
    const tree = renderer.create(<CheckboxInput question={defaultQuestion} />)
    expect(tree).toMatchSnapshot()
  })
})
