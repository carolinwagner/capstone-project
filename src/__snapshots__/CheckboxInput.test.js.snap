// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxInput renders CheckboxInput correctly 1`] = `
<div>
  <div>
    <input
      className="CheckboxInput__StyledCheckboxInput-sc-131g3tm-0 brfjmz"
      id="a"
      name="undefined[0]"
      type="checkbox"
      value="a"
    />
    <label
      htmlFor="a"
    >
      a
    </label>
  </div>
  <div>
    <input
      className="CheckboxInput__StyledCheckboxInput-sc-131g3tm-0 brfjmz"
      id="b"
      name="undefined[1]"
      type="checkbox"
      value="b"
    />
    <label
      htmlFor="b"
    >
      b
    </label>
  </div>
  <div>
    <input
      className="CheckboxInput__StyledCheckboxInput-sc-131g3tm-0 brfjmz"
      id="d"
      name="undefined[2]"
      type="checkbox"
      value="c"
    />
    <label
      htmlFor="d"
    >
      c
    </label>
  </div>
  <div>
    <input
      className="CheckboxInput__StyledCheckboxInput-sc-131g3tm-0 brfjmz"
      id="d"
      name="undefined[3]"
      type="checkbox"
      value="d"
    />
    <label
      htmlFor="d"
    >
      d
    </label>
  </div>
</div>
`;
